src/main.ts

```diff
@@ -12,21 +20,27 @@ const networkConfigUrl = (() => {
   }
 })();
 
-const leader = makeLeader(networkConfigUrl);
-const follower = makeFollower(':published.vaultFactory.governance', leader);
-
-type Params = { current: { EndorsedUI: { type: 'string'; value: string } } };
-
 const setMessage = (message: string) => {
   document?.getElementById('msg')?.replaceChildren(message);
   console.info(message);
 };
 
 const tryRedirect = async () => {
+  const rpc = await fetchRPCAddr(networkConfigUrl);
   let endorsedUI;
-  for await (const { value } of iterateLatest<{ value: Params }>(follower)) {
-    endorsedUI = value.current.EndorsedUI;
-    break;
+
+  try {
+    const data = await abciQuery(
+      rpc,
+      '/custom/vstorage/data/published.vaultFactory.governance'
+    );
```

Comment:
consider a `vstorageQuery` function that takes `published.vaultFactory.governance`.